src/wallet/api/wallet.cpp

```diff
@@ -2819,4 +2875,853 @@ uint64_t WalletImpl::getBytesSent()
     return m_wallet->get_bytes_sent();
 }
 
+//-------------------------------------------------------------------------------------------------------------------
+std::string WalletImpl::getMultisigSeed(const std::string &seed_offset) const
+{
+    clearStatus();
+
+    try
+    {
+        checkMultisigWalletReady(m_wallet);
+
+        epee::wipeable_string seed;
+        if (m_wallet->get_multisig_seed(seed, seed_offset))
+            return std::string(seed.data(), seed.size());
+    }
+    catch (const std::exception &e)
+    {
+        LOG_ERROR(__FUNCTION__ << " error: " << e.what());
+        setStatusError(string(tr("Failed to get multisig seed: ")) + e.what());
+    }
+    return "";
+}
+//-------------------------------------------------------------------------------------------------------------------
+std::pair<std::uint32_t, std::uint32_t> WalletImpl::getSubaddressIndex(const std::string &address) const
+{
+    clearStatus();
+
+    cryptonote::address_parse_info info;
+    std::pair<std::uint32_t, std::uint32_t> indices{0, 0};
+    if (!cryptonote::get_account_address_from_str(info, m_wallet->nettype(), address))
+    {
+        setStatusError(string(tr("Failed to parse address: ") + address));
+        return indices;
+    }
+
+    auto index = m_wallet->get_subaddress_index(info.address);
+    if (!index)
+        setStatusError(string(tr("Address doesn't belong to the wallet: ") + address));
+    else
+        indices = std::make_pair((*index).major, (*index).minor);
+
+    return indices;
+}
+//-------------------------------------------------------------------------------------------------------------------
+void WalletImpl::freeze(std::size_t idx)
+{
+    clearStatus();
+
+    try
+    {
+        m_wallet->freeze(idx);
+    }
+    catch (const std::exception &e)
+    {
+        LOG_ERROR(__FUNCTION__ << " error: " << e.what());
+        setStatusError((boost::format(tr("Failed to freeze enote with index `%zu`: %s")) % idx % e.what()).str());
+    }
+}
+//-------------------------------------------------------------------------------------------------------------------
+void WalletImpl::freeze(const std::string &key_image)
```

Comment:
Keep in mind that key images are not always available (e.g. in view-only wallets). Referencing enotes by pubkey works consistently.